runtime-config-linux.md

```diff
@@ -146,45 +146,128 @@ The cgroups will be created if they don't exist.
 
 `cgroupsPath` can be used to either control the cgroups hierarchy for containers or to run a new process in an existing container.
 
-Optionally, cgroups limits can be specified via `resources`.
+You can configure a container's cgroups via the `resources` field of the Linux configuration.
+Do not specify `resources` unless limits have to be updated.
+For example, to run a new process in an existing container without updating limits, `resources` need not be specified.
+
+#### Disable out-of-memory killer
 
 ```json
-    "resources": {
-        "disableOOMKiller": false,
-        "memory": {
-            "limit": 0,
-            "reservation": 0,
-            "swap": 0,
-            "kernel": 0,
-            "swappiness": -1
-        },
-        "cpu": {
-            "shares": 0,
-            "quota": 0,
-            "period": 0,
-            "realtimeRuntime": 0,
-            "realtimePeriod": 0,
-            "cpus": "",
-            "mems": ""
-        },
-        "blockIO": {
-            "blkioWeight": 0,
-            "blkioWeightDevice": "",
-            "blkioThrottleReadBpsDevice": "",
-            "blkioThrottleWriteBpsDevice": "",
-            "blkioThrottleReadIopsDevice": "",
-            "blkioThrottleWriteIopsDevice": ""
-        },
-        "hugepageLimits": null,
-        "network": {
-            "classId": "",
-            "priorities": null
-        }
-    }
+	"disableOOMKiller": false
 ```
 
-Do not specify `resources` unless limits have to be updated.
-For example, to run a new process in an existing container without updating limits, `resources` need not be specified.
+#### Memory
+
+```json
+	"memory": {
+		"limit": 0,
+		"reservation": 0,
+		"swap": 0,
+		"kernel": 0,
+		"swappiness": -1
+	}
+```
+
+#### CPU
+
+```json
+	"cpu": {
+		"shares": 0,
+		"quota": 0,
+		"period": 0,
+		"realtimeRuntime": 0,
+		"realtimePeriod": 0,
+		"cpus": "",
+		"mems": ""
+	}
+```
+
+#### Block IO Controller
+
+`blockIO` represents the cgroup subsystem `blkio` which implements the block io controller.
+For more information, see the [kernel cgroups documentation about `blkio`](https://www.kernel.org/doc/Documentation/cgroups/blkio-controller.txt).
+
+```json
+	"blockIO": {
+		"blkioWeight": 0,
+		"blkioLeafWeight": 0,
+		"blkioWeightDevice": [
+			{
+				"major": 8,
+				"minor": 0,
+				"weight": 500,
+				"leafWeight": 300
+			},
+			{
+				"major": 8,
+				"minor": 16,
+				"weight": 500
+			}
+		],
+		"blkioThrottleReadBpsDevice": [
+			{
+				"major": 8,
+				"minor": 0,
+				"rate": 600
+			}
+		],
+		"blkioThrottleWriteBpsDevice": null,
+		"blkioThrottleReadIOPSDevice": null,
+		"blkioThrottleWriteIOPSDevice": [
+			{
+				"major": 8,
+				"minor": 16,
+				"rate": 300
+			}
+		]
+	}
+```
+
+* **blkioWeight** (uint16, optional) - specifies per cgroup weight, this is default weight of the group on all the devices until and unless overridden by per device rule, range is from 10 to 1000
```

Comment:
And probably:

* “per cgroup” → “per-cgroup”
* “, this is default” → “.  This is the default”
* “all the devices” → “all devices”
* “per device rule” → “per-device rules”.
* “, range is from” → “.  The range is from”

I'd consider leaving the range description out (and just expecting folks to get it from the kernel docs) so we don't go stale in the unlikely event that the kernel changes the range.